lib/package_source.py

```diff
@@ -10,6 +10,26 @@
 from lib import utils
 
 
+def set_dest_dir(source, directory, local_copy_subdir_name):
+
+    source_type = source.keys()[0]
+    if source_type != 'url':
+        if source_type == 'svn':
+            dest = os.path.join(directory, local_copy_subdir_name)
+        else:
+            repo_name = os.path.basename(source[source_type]['src'])
+            dest = os.path.join(directory, repo_name)
+    else:
+        file_name = os.path.basename(source['url']['src'])
+        dest = os.path.join(directory, file_name)
+
+    source[source_type]['dest'] = dest
+
+    # in git, git repo object must be created for later git archive
+    if source_type == 'git':
```

Comment:
- It seems to me that, currently, this will be only run once for each package in each host_os.py script execution
- If for some reason, it starts to run more than once, the 'src' attribute value in source may have changed since last execution and IMHO the Git repository object must be recreated